packages/dds/sequence/src/test/sharedString.spec.ts

```diff
@@ -236,6 +236,36 @@ describe("SharedString", () => {
 			assert.equal(simpleMarker.properties?.color, "blue", "Could not annotate marker");
 		});
 
+		it("fails when the marker is updated", () => {
+			sharedString.insertText(0, "hello world");
+			// Insert a simple marker.
+			sharedString.insertMarker(6, ReferenceType.Simple, {
+				[reservedMarkerIdKey]: "markerId",
+			});
+			// Annotate the marker.
+			const props = { color: "blue" };
+			const simpleMarker = sharedString.getMarkerFromId("markerId") as Marker;
+			sharedString.annotateMarker(simpleMarker, props);
+			assert.equal(simpleMarker.properties?.color, "blue", "Could not annotate marker");
+			// Annotate the marker's ID.
+			const newIdProps = { [reservedMarkerIdKey]: "newIdValue" };
+			try {
+				sharedString.annotateMarker(simpleMarker, newIdProps);
+				assert.fail();
+			} catch (error) {
+				assert.notEqual(
+					error,
+					undefined,
+					"Error from attempting to update marker was not thrown",
+				);
+				assert.equal(
+					(error as Error).message,
+					"Cannot change the markerId of an existing marker",
+					`Incorrect error received - got ${(error as Error).message}`,
+				);
```

Comment:
The intention here is correct, but this way of checking it will break during the next release because our tooling will replace the error message in the `assert()` in `sharedString.ts` with a hex code, so this check will stop seeing the string message. You can use `validateAssertionError` from `@fluidframework/test-runtime-utils` to do it and it'll keep working through that replacement (example [here](https://github.com/microsoft/FluidFramework/blob/a61bd2d371642c3dcadfcdb1ab501faa101e5849/packages/dds/shared-object-base/src/test/sharedObject.spec.ts#L95)).